### ISSUE TITLE
Lacona should "learn" what apps you launch

### ISSUE THREAD

**Issue by Nxt3** [laconalabs/LaconaApp#48]
I don't know how feasible this is. But after launching the same apps over and over again in Spotlight--it starts to suggest it more quickly. Good example of this would be Google Keep. I have two apps, "KeepingYouAwake" and "Google Keep". If I type in "Kee" in Spotlight, it suggests Google Keep. Typing in "Kee" in Lacona yields "KeepingYouAwake". I'm not sure if Spotlight learns--but I launch Keep on a daily basis, and KeepingYouAwake almost never.

**Comment by brandonhorst** [comment]
It's feasible but it's difficult to do correctly. Because Lacona is more complex then just command/argument syntax (or only-argument syntax in Spotlight's case), there are design issues that come up.

Just as some examples:
- Let's say you "open sublime text" 10 times and you "quit skype" 20 times. Should typing "open s" pull up sublime text or skype? That is to say, do you only track the number of times an "application" has been used, or how many times a particular command in its entirety has been used? If you say the latter, then what about more complex cases? "open todo.txt in s", or "open facetime and s".
- If you "call Vicky" quite a bit, should "Vicky's birthday" sort above "Vivian's birthday" when it comes to dates?

Another thing I want to avoid is the "thrashing" effect when you use two similar commands almost-equally. When you don't know at all which one is going to sort first, then it might as well not autocomplete at all, because you need to pay careful attention every time you run the command.

These questions have answers and the problems are solvable, but they're not easy. When it comes to features like this, I would rather have them not exist at all than be implemented poorly.

tl;dr this is on the radar, but it isn't going to happen immediately.

**Comment by Nxt3** [comment]
You make valid points; never thought of it like that. Even so--I'm looking forward to the future of this app. :+1:

**Comment by kopischke** [comment]
@brandonhorst excellent points; the thrashing effect is one of the things that has turned me off of Alfred.

**Comment by madbonkey** [comment]
An alternative (or additional feature?) would be to add alias-like functionality, so we can add custom shortcuts through the preferences.
